File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
    List<Integer> l=new ArrayList<Integer> ();
        
        traversal(l,root);
        return l;
        
    }
    
    public void traversal(List<Integer> l , TreeNode root)
    {
        if(root==null)
            
            return;
        
        traversal(l, root.left);
        l.add(root.val);
        traversal(l,root.right);
    }
    
}

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        list1=[]
        self.traversal(root,list1)
        return list1
    def traversal(self,root,list1):
        if  root==None:
            return
        self.traversal(root.left,list1)
        list1.append(root.val)
        self.traversal(root.right,list1) 
        